### PR TITLE
Add documentation for Streaming API

### PR DESCRIPTION
**Pull request by owentmfoo**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
-interfaces with any viewer, network protocol, or storage technology of their choice. 
-All the data from the SECU unit (after RDA filtering) will be available on the stream.
-This will be integrated with a protocol for streaming engineering (calibrated) 
-telemetry, interoperating with ATLAS clients and the surrounding data processing 
-ecosystem.
-
-[Stream API / Protocol](stream_api/index.md) 
+## [Open Streaming Architecture](open_streaming_architecture/index.md)
+A standard, open set of rules and services for streaming telemetry data over a broker 
+(assumed to be Kafka, but other brokers will be supported). Includes full definition of
```
Comment (harveyAJ) [inline docs/index.md:12]: Not any broker, they need to support the concept of topic etc. Will we support other brokers? Probably a question for Nikhil

File: docs/open_streaming_architecture/index.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Open Streaming Architecture
+A standard, open set of rules and services for streaming telemetry data over a broker 
+(assumed to be Kafka, but other brokers will be supported). Includes full definition of 
+the Protobuf schema for transmitting messages, along with documentation describing the 
+messages.
+ 
+**Hight level pic**
```
Comment (harveyAJ) [inline docs/open_streaming_architecture/index.md:7]: ?

File: docs/open_streaming_architecture/index.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Open Streaming Architecture
+A standard, open set of rules and services for streaming telemetry data over a broker
```
Comment (harveyAJ) [inline docs/open_streaming_architecture/index.md:2]: We probably don't need to repeat this paragraph copied/pasted from the parent page?

File: docs/open_streaming_architecture/index.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ 
+**Hight level pic**
+ 
+
+## Components
+
+### [Stream API](stream_api/overview)
+A standard API to expose streaming data from SECU units and abstract away proprietary 
+implementation details. Consumers of this API will be able to write software that 
+interfaces with any viewer, network protocol, or storage technology of their choice.
```
Comment (harveyAJ) [inline docs/open_streaming_architecture/index.md:15]: What do we mean by "network protocol"

File: docs/open_streaming_architecture/stream_api/overview.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Stream API
+
+The Stream API is responsible for publishing and consuming data from the Kafka broker. Created under the 
+[gRPC framework](https://grpc.io/), it allows clients to communicate to a server which handles the Kafka communication.
+
+The server and the client can be deployed locally within the same application or the server can be deployed remotely 
+with the docker image provided.
```
Comment (harveyAJ) [inline docs/open_streaming_architecture/stream_api/overview.md:7]: Would be nice to add a hyperlink to Dockerhub here

File: docs/open_streaming_architecture/stream_api/overview.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+Three components are currently available for evaluation:
+
+1. Stream API server as docker image
```
Comment (harveyAJ) [inline docs/open_streaming_architecture/stream_api/overview.md:11]: Again, include links to relevant packages/images

File: docs/open_streaming_architecture/stream_api/overview.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+Three components are currently available for evaluation:
+
+1. Stream API server as docker image 
+2. Stream API server as NuGet package 
+3. Stream API client as NuGet package
+
+## Batching Responses
+
+When a high amount of messages is processed at the same time. The overhead can be significant and the bandwidth can be
```
Comment (harveyAJ) [inline docs/open_streaming_architecture/stream_api/overview.md:17]: should be a comma after "time"

File: docs/open_streaming_architecture/stream_api/overview.md (hunk 6 above)
Comment (owentmfoo) [reply docs/open_streaming_architecture/stream_api/overview.md]: Link added

File: docs/open_streaming_architecture/stream_api/overview.md (hunk 7 above)
Comment (owentmfoo) [reply docs/open_streaming_architecture/stream_api/overview.md]: fixed

File: docs/open_streaming_architecture/stream_api/overview.md (hunk 5 above)
Comment (owentmfoo) [reply docs/open_streaming_architecture/stream_api/overview.md:7]: link added to the component listed below instead.

**Review by nikhilsawant22 — APPROVED** [review]

**Pull request merged**